Update BBEs with default payload param support (#4119)
ballerina-platform/ballerina-distribution: examples/http-query-parameter/http_query_parameter.md

# REST service - Query parameter

> The query parameter in the resource argument represents the query segment of the request URL. The argument name should be the key of the query, and its value is mapped during the runtime by extracting it from the URL. Except for resolving ambiguity with the default payload parameter, no additional annotation is required for the query parameter. However, the `@http:Query` annotation can be used in such cases. The supported types are `string`, `int`, `float`, `boolean`, `decimal`, `map<json>`, and `array` types of the aforementioned types. The query parameter type can be nilable (e.g., (`string? bar`)) and defaultable (e.g., (`string bar = "hello"`)). When a request contains query segments, retrieving them as resource arguments is much simpler and well-recommended. Alternatively, the `http:Request` also provides related methods to retrieve query parameters.

Reviewer: I think we can revert these text changes. We can add it later as a note if needed. `@http:Query` is need only in extreme edge cases.

Reviewer 2: Fixed

> The query parameter in the resource argument represents the query segment of the request URL. The argument name should be the key of the query, and its value is mapped during the runtime by extracting it from the URL. Except for resolving ambiguity with the default payload parameter, no additional annotation is required for the query parameter. However, the `@http:Query` annotation can be used in such cases. The supported types are `string`, `int`, `float`, `boolean`, `decimal`, `map<json>`, and `array` types of the aforementioned types. The query parameter type can be nilable (e.g., (`string? bar`)) and defaultable (e.g., (`string bar = "hello"`)). When a request contains query segments, retrieving them as resource arguments is much simpler and well-recommended. Alternatively, the `http:Request` also provides related methods to retrieve query parameters.

Reviewer 2: ```suggestion
The query parameter in the resource argument represents the query segment of the request URL. The argument name should be the key of the query, and its value is mapped during the runtime by extracting it from the URL. The query parameter does not need any additional annotation. The supported types are `string`, `int`, `float`, `boolean`, `decimal`, and `array` types of the aforementioned types. The query parameter type can be nilable (e.g., (`string? bar`)) and defaultable (e.g., (`string bar = "hello"`)). When a request contains query segments, retrieving them as resource arguments is much simpler and well-recommended. Alternatively, the `http:Request` also provides related methods to retrieve query parameters.
```

Reviewer 2: Fixed via [fcd7eff](https://github.com/ballerina-platform/ballerina-distribution/pull/4119/commits/fcd7eff8fab4660b16bfcb622b4bbf9504761107)
